crates/rpc/tests/rpc.rs

```diff
@@ -0,0 +1,488 @@
+        expected: &BlockWithTxs,
+    ) {
+        let transaction = client.get_transaction_by_block_id_and_index(block_id, transaction_index as u64).await.expect("Failed to retrieve a specific transaction by its block ID and index");
+
+        // `starknet-rs` doesn't implement `PartialEq` on its DTOs, and transactions have many *variants which makes pure Rust comparison painful.
+        // Just serialize these to compare them.
+        assert_eq!(
+            serde_json::to_string(&transaction).unwrap(),
+            serde_json::to_string(&expected.transactions[transaction_index])
+                .unwrap(),
+        )
+    }
+
+    for transaction_index in 0..10 {
+        check_transaction(&client, block_id, transaction_index, &block).await;
+    }
+
+    check_transaction(&client, block_id, block.transactions.len() - 1, &block)
+        .await;
+}
+
+#[tokio::test]
+async fn test_get_transaction_by_hash() {
+    let TestContext { client, block, block_id: _, extracted_value: () } =
+        latest_10_tx_context().await;
+
+    async fn check_transaction(
+        client: &JsonRpcClient<HttpTransport>,
+        expected: &Transaction,
+    ) {
+        let transaction = client.get_transaction_by_hash(expected.transaction_hash()).await.expect("Failed to retrieve a specific transaction by its block ID and index");
+
+        // `starknet-rs` doesn't implement `PartialEq` on its DTOs, and transactions have many *variants which makes pure Rust comparison painful.
+        // Just serialize these to compare them.
+        assert_eq!(
+            serde_json::to_string(&transaction).unwrap(),
+            serde_json::to_string(&expected).unwrap(),
+        )
+    }
+
+    for transaction_index in 0..10 {
+        check_transaction(&client, &block.transactions[transaction_index])
+            .await;
+    }
+
+    check_transaction(
+        &client,
+        block
+            .transactions
+            .last()
+            .as_ref()
+            .expect("We need a last transaction here"),
+    )
+    .await;
+}
+
+#[tokio::test]
+async fn test_get_transaction_status() {
+    let TestContext { client, block, block_id: _, extracted_value: () } =
+        latest_10_tx_context().await;
+
+    async fn check_transaction(
+        client: &JsonRpcClient<HttpTransport>,
+        transaction_hash: FieldElement,
+    ) {
+        let _status = client
+            .get_transaction_status(transaction_hash)
+            .await
+            .expect("Failed to retrieve the transaction status");
+    }
+
+    for transaction_index in 0..10 {
+        check_transaction(
+            &client,
+            *block.transactions[transaction_index].transaction_hash(),
+        )
+        .await;
+    }
+
+    check_transaction(
+        &client,
+        *block
+            .transactions
+            .last()
+            .as_ref()
+            .expect("We need a last transaction here")
+            .transaction_hash(),
+    )
+    .await;
+}
+
+/* TODO
```

Comment:
seems like a good candidate to be extracted as an issue :)

Reply:
Will do once merged